File: MK/10월 3주차/행렬의곱셈.java
class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {

        int n=arr1.length;
        int m=arr2[0].length;
        int c=arr2.length;

        int[][] res = new int[n][m];

        for (int i=0; i<n; i++) {
            for (int j=0; j<m; j++) {
                int tmp=0;
                for (int k=0; k<c;k++) {
                    tmp+= arr1[i][k] * arr2[k][j];
                }
                res[i][j]=tmp;
            }
        }

        return res;
    }
}

File: BW/10월3주차/행렬의곱셈.js
function solution(arr1, arr2) {
    var col = arr1.length; // arr1의 세로 길이
    var row = arr2[0].length; // arr2의 가로 길이
    var len = arr2.length; // arr2의 세로 길이
    var newArr = new Array(col);   

    for (var i = 0; i < col; i++) {
        newArr[i] = new Array(row); // 2차원 배열 선언
        for (var j = 0; j < row; j++) {
            var sum = 0;
            for (var k = 0; k < len; k++) {
                sum += arr1[i][k] * arr2[k][j]; 
            }
            newArr[i][j] = sum; // 결과 계산
        }
    }
    return newArr;
}
